backends/xnnpack/cmake/Dependencies.cmake

```diff
@@ -10,71 +10,83 @@ set(THIRD_PARTY_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/third-party")
 )
-set(XNNPACK_BUILD_TESTS
-    OFF
-    CACHE BOOL ""
+get_extra_cmake_args_for_external_project(XNNPACK_EXTRA_CMAKE_ARGS)
+ExternalProject_Add(
+  XNNPACKExternalProject
+  SOURCE_DIR ${XNNPACK_SOURCE_DIR}
+  # Not 100% clear on these locations
+  BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/XNNPACK
+  INSTALL_DIR ${XNNPACK_INSTALL_DIR}
+  INSTALL_BYPRODUCTS ${XNNPACK_STATIC_LIB} ${XNNPACK_MICROKERNELS_STATIC_LIB}
+  CMAKE_ARGS
+    ${XNNPACK_EXTRA_CMAKE_ARGS}
+    -D
+    XNNPACK_LIBRARY_TYPE=static
+    -D
+    XNNPACK_BUILD_BENCHMARKS=OFF
+    -D
+    XNNPACK_BUILD_TESTS=OFF
+    -D
+    XNNPACK_ENABLE_AVXVNNI=OFF
+    # Work around observed failure:
+    # https://github.com/pytorch/executorch/pull/10362#issuecomment-2906391232
+    -D
+    XNNPACK_ENABLE_AVX512VNNIGFNI=OFF
+    -D
+    ENABLE_XNNPACK_WEIGHTS_CACHE=${EXECUTORCH_XNNPACK_ENABLE_WEIGHT_CACHE}
+    -D
+    ENABLE_XNNPACK_SHARED_WORKSPACE=${EXECUTORCH_XNNPACK_SHARED_WORKSPACE}
+    -D
+    XNNPACK_ENABLE_KLEIDIAI=${EXECUTORCH_XNNPACK_ENABLE_KLEIDIAI}
+    -D
+    CMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -D
+    XNNPACK_BUILD_ALL_MICROKERNELS=OFF
+    -D
+    CMAKE_POSITION_INDEPENDENT_CODE=ON
 )
-set(XNNPACK_ENABLE_AVXVNNI
-    OFF
-    CACHE BOOL ""
-  )
-# Work around observed failure: https://github.com/pytorch/executorch/pull/10362#issuecomment-2906391232
-set(XNNPACK_ENABLE_AVX512VNNIGFNI
-  OFF
-  CACHE BOOL "")
 
-if(EXECUTORCH_XNNPACK_ENABLE_KLEIDI)
-    set(XNNPACK_ENABLE_KLEIDIAI
-        ON
-        CACHE BOOL ""
-    )
-else()
-    set(XNNPACK_ENABLE_KLEIDIAI
-        OFF
-        CACHE BOOL ""
-    )
-endif()
+add_library(XNNPACK STATIC IMPORTED GLOBAL)
+# TODO: this probably doesn't work on Windows.
+set_property(TARGET XNNPACK PROPERTY IMPORTED_LOCATION ${XNNPACK_STATIC_LIB})
 
+add_dependencies(XNNPACK XNNPACKExternalProject)
 
-set(XNNPACK_BUILD_ALL_MICROKERNELS
-    OFF
-    CACHE BOOL ""
+add_library(xnnpack-microkernels-prod STATIC IMPORTED GLOBAL)
+set_property(
+  TARGET xnnpack-microkernels-prod PROPERTY IMPORTED_LOCATION
+                                            ${XNNPACK_MICROKERNELS_STATIC_LIB}
 )
-add_subdirectory("${XNNPACK_SOURCE_DIR}")
-include_directories(SYSTEM ${XNNPACK_INCLUDE_DIR})
-list(APPEND xnnpack_third_party XNNPACK)
-install(TARGETS xnnpack-microkernels-prod
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+add_dependencies(xnnpack-microkernels-prod XNNPACKExternalProject)
 
+set_target_properties(
+  XNNPACK PROPERTIES INTERFACE_LINK_LIBRARIES xnnpack-microkernels-prod
+)
+
+install(DIRECTORY ${XNNPACK_INSTALL_DIR}/ DESTINATION ${CMAKE_INSTALL_PREFIX})
 
 if(EXECUTORCH_XNNPACK_ENABLE_KLEIDI)
-    install(TARGETS kleidiai
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+  add_library(kleidiai SHARED IMPORTED)
```

Comment:
i think i merged something that might have conflict here. I just added a check 

if (TARGET kleidiai)

before adding the library